src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt

```diff
@@ -95,8 +97,16 @@ abstract class BaseGenerateHandler : LanguageCodeInsightActionHandler {
         struct: RsStructItem
     ): RsImplItem {
         return if (implBlock == null) {
-            val impl = psiFactory.createInherentImplItem(structName, struct.typeParameterList, struct.whereClause)
-            struct.parent.addAfter(impl, struct) as RsImplItem
+            val existingImpl = struct.implLookup.findImplsAndTraits(struct.declaredType)
+                .filterIsInstance<TraitImplSource.ExplicitImpl>()
+                .filter { it.isInherent }
+                .firstOrNull()
```

Comment:
This `firstOrNull` looks very suspicious. A struct can have multiple inherent impls. Or impls with different type bounds... There can be one another dialog that allows to choose the impl (or choose to create a new one). But a new dialog overloads the UI, so such a solution is also very discussable. I'd prefer to generate a new impl over this design.

Reply:
Oh right, I didn't think of impls with different type bounds. Ok, let's ditch finding existing impls. Should I also remove it in https://github.com/intellij-rust/intellij-rust/pull/6039?

Reply:
Yep, I just reviewed #6039 and was just about to write a comment about it =D